index.js

```diff
@@ -56,37 +58,41 @@ const query = `
 	}
 `;
 
-let responseText = '[]';
 let responseETag = '';
 
-async function fetchRepos() {
-	const {body} = await graphqlGot('api.github.com/graphql', {
-		query,
-		token
-	});
-
-	const repos = body.user.repositories.nodes.map(repo => ({
-		...repo,
-		stargazers: repo.stargazers.totalCount,
-		forks: repo.forks.totalCount
-	}));
-	responseText = JSON.stringify(repos);
-	responseETag = etag(responseText);
-}
-
-setInterval(fetchRepos, ONE_DAY);
-fetchRepos();
-
-module.exports = (request, response) => {
+module.exports = async (request, response) => {
 	controlAccess()(request, response);
-	response.setHeader('cache-control', cache);
-	response.setHeader('etag', responseETag);
 
 	if (request.headers.etag === responseETag) {
 		response.statusCode = 304;
 		response.end();
 		return;
 	}
 
-	response.end(responseText);
+	try {
+		const {body} = await graphqlGot('api.github.com/graphql', {
+			query,
+			token: GITHUB_TOKEN
+		});
+
+		const repos = JSON.stringify(
+			body.user.repositories.nodes.map(repo => ({
+				...repo,
+				stargazers: repo.stargazers.totalCount,
+				forks: repo.forks.totalCount
+			}))
+		);
+
+		responseETag = etag(repos);
+
+		response.setHeader('cache-control', `s-maxage=${ONE_DAY}, max-age=${CACHE_MAX_AGE}`);
+		response.setHeader('etag', responseETag);
+		response.end(repos);
+	} catch (error) {
+		console.error(error);
+
+		response.statusCode = 500;
+		response.setHeader('content-type', 'text/plain');
+		response.end('Internal server error');
```

Comment:
I'm surprised the Now builder doesn't handle this for us, if the exported function throws.

Reply:
Now does have a generic error screen, but it gives `502` and says "No response from Lambda" or something.